Header header
string action
int8 action_type